modules/mono/csharp_script.cpp

```diff
@@ -297,6 +297,20 @@ Ref<Script> CSharpLanguage::get_template(const String &p_class_name, const Strin
 	return script;
 }
 
+bool CSharpLanguage::is_using_templates() {
+
+	return true;
+}
+
+void CSharpLanguage::make_template(const String &p_class_name, const String &p_base_class_name, Ref<Script> &p_script) {
+
+	String src = p_script->get_source_code();
+	src = src.replace("%BASE%", p_base_class_name)
+				  .replace("%CLASS%", p_class_name)
+				  .replace("%TS%", _get_indentation());
```

Comment:
The default indent type will be tabs like for GDScript, but the current default script template uses spaces: https://github.com/Hinsbart/godot/blob/f3218c24c7735ce7497356dfa82f8a3d67981d0b/modules/mono/csharp_script.cpp#L265-L287

Reply:
We have 3 options IMO:
- Hard-code it to use only spaces.
- Make it use global editor settings for indentation.
- Override editor settings for C# like we do for external editor, with 3 options: Use global editor settings, use spaces, use tabs.

Reply:
I'd say option 3 would be the best. I can see myself using tabs for GDScript since I got used to, and spaces for C# as it's more conventional.